lib/fetch/headers.js

```diff
@@ -528,8 +528,10 @@ Object.defineProperties(Headers.prototype, {
 
 webidl.converters.HeadersInit = function (V) {
   if (webidl.util.Type(V) === 'Object') {
-    if (V[Symbol.iterator]) {
```

Comment:
So there are two bugs here:
- The existence of a Symbol.iterator does not mean a value is a sequence.
- Accessing Symbol.iterator this way is detectable via Proxy. Once we pass V to the sequence converted it once again accesses V[Symbol.iterator].